documentation/docs/scenarios/upload-to-transportrequest/Readme.md

```diff
@@ -0,0 +1,197 @@
+
+The transport request can be created on the fly (see [transportRequestCreate](../../../steps/transportRequestCreate/)) or we can use an already existing transport request. In case we use an already existing transport request Id the transport request Id needs to be provided in the git commit history (see example below) or the transport request id needs to be provided inside the job (e.g. as a job parameter).
+
+The transport request can be closed by the pipeline job (see [transportRequestRelease](../../../steps/transportRequestRelease/)))
+
+A git commit message containing the transportRequestId:
+
+```
+The headline
+
+The body. The blank line above is mandatory (git standard).
+TransportRequest: <YOUR TRANSPORT REQUEST ID>
+```
+
+By default the git commits between the merge base with base branch (default: `master`) and the current branch head are traversed.
+
+![This pipeline in Jenkins Blue Ocean](images/pipeline.png)
+###### Screenshot: Build and Deploy Process in Jenkins
+
+## Examples
+
+### Upload via RFC
+
+#### Jenkinsfile
+
+Following the convention for pipeline definitions, use a `Jenkinsfile` which resides in the root directory of your development sources.
+
+```groovy
+@Library('piper-lib-os') _
+
+
+pipeline {
+
+    agent any
+
+    stages {
+        stage("prepare") {
+            steps {
+                deleteDir()
+                checkout scm
+                setupCommonPipelineEnvironment script: this
+            }
+        }
+
+        stage('build') {
+            steps {
+                // what needs to be done here depends on the project, maybe zipping the sources is sufficient
+                mtaBuild script: this
+            }
+        }
+
+        stage('publish') {
+            steps {
+                // uploads the binary into a blob store so that it can be attached to a transport request later
+                sh "curl --upload-file <deployable> <BLOB_STORE/path/to/application>"
+
+                // OR (in case there is no BLOB_STORE available)
+
+                // this makes the artifact available on nexus. The url is:
+                // <JENKINS_URL>/job/<JOB_NAME>/<BUILD_NUMBER>/artifact/<DEPLOYABLE>. Nota bene: this format is not an Jenkins API.
+                // The build number can be retrieved during the build via ${currentBuild.number}
+                archiveArtifacts artifacts: <deployable>
+            }
+        }
+
+        // attaches the deployable to a transport request
+        stage('attach') {
+            steps {
+                transportRequestUploadFile script: this,
+                                           transportRequestId: '<TRANSPORT_REQUEST_ID>', // can be omitted if present inside a git commit message
```

Comment:
```suggestion
                                           transportRequestId: '<TRANSPORT_REQUEST_ID>', // This can be omitted if present inside a Git commit message
```